### ISSUE TITLE
Inconsistency in C++ cross_entropy_loss and other loss terms

### ISSUE THREAD

**Issue by EmmettBicker** [pytorch/pytorch#142146]
### 🐛 Describe the bug

Looking at the method arguments In the bindings generated in python_nn_functions.cpp, almost all loss functions have an out parameter but cross_entropy_loss

```cpp
cross_entropy_loss(
    Tensor input, 
    Tensor target, Tensor? 
    weight=None, 
    int64_t reduction=at::Reduction::Mean, 
    SymInt ignore_index=-100, 
    double label_smoothing=0.0
)
```
```cpp
mse_loss(
    Tensor input, 
    Tensor target, 
    int64_t reduction=at::Reduction::Mean, 
    *, 
    Tensor out=None
)
```
```cpp
binary_cross_entropy(
    Tensor input, 
    Tensor target, 
    Tensor? weight=None, 
    int64_t reduction=at::Reduction::Mean, 
    *, 
    Tensor out=None
)

I haven't found a loss function without it except for cross_entropy and the following loss functions do have the out parameter: binary_cross_entropy, l1_loss, smooth_l1_loss, max_pool2d_with_indices, max_pool3d_with_indices, huber_loss, mse_loss, multilabel_margin_loss, soft_margin_loss, multi_margin_loss 

There might be an intentional reason for this, but I thought I should bring it up since it looked inconsistent.

### Versions

'2.6.0a0+git7224cd4'
This isn't version specific so I'm going to omit further versions info

**Comment by mikaylagawarecki** [comment]
Is this specific to the cpp API or the python bindings in `nn.functional.`? I don't see the out arguments being exposed for the loss functions in `nn.functional`

**Comment by EmmettBicker** [comment]
I was looking at the cpp bindings in torch._C._nn, but those are manually created and a lot of them miss the out parameter. I’m working on this PR to autogenerate the stubs which is where I found the discrepancy #142049
